fe/fe-core/src/main/java/org/apache/doris/nereids/memo/Memo.java

```diff
@@ -465,6 +468,8 @@ public Group mergeGroup(Group source, Group destination) {
             source.moveOwnership(destination);
             groups.remove(source.getGroupId());
         }
+
+        needMergeGroups.forEach(this::mergeGroup);
```

Comment:
consider a scenario:
if we have two entry in the map: B -> A and C -> B.
when we do merge group. we first merge B to A, and then B is removed. after that, we want to merge C to B, but B is not exist anymore.
So, maybe we need to use **disjoint set** do find a common parent, and then merge all other group to it in one **disjoint set**.